standardtracer/propagation.go

```diff
@@ -177,52 +181,48 @@ func (p *splitBinaryPropagator) JoinTrace(
 	var traceID, propagatedSpanID int64
 	var sampledByte byte
 
-	err = binary.Read(contextReader, binary.BigEndian, &traceID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &traceID); err != nil {
 		return nil, opentracing.ErrTraceCorrupted
 	}
-	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &propagatedSpanID); err != nil {
```

Comment:
if we decide to keep standardtracer around (and I am tempted, seeing the great work @tschottdorf is doing), this propagation logic should be in some other package, because it makes explicit assumption about the shape of tracer context being `(trace-id, span-id, flags)`, which is Dapper-ish, but not Zipkin-like.